lib/src/api.ts

```diff
@@ -28,16 +45,20 @@ export type PayloadRequestData = {
   quoteId?: string;
   toNewTokenId?: string;
 };
-export type PayloadResponse = {
+export type SwapPayloadResponse = {
   binaryPayload: string;
   signature: string;
   payinAddress: string;
   swapId: string;
   payinExtraId?: string;
 };
-export async function retrievePayload(
-  data: PayloadRequestData,
-): Promise<PayloadResponse> {
+
+export async function retriveSwapPayload(
+  data: SwapPayloadRequestData
+): Promise<SwapPayloadResponse> {
+  // Make sure we are using the correct URL for the BE
+  setBackendUrlAsSwap();
+
```

Comment:
This does not make sure you're using the correct URL. This does the exact opposite 😹 I should remove the ENV with the prod URL in exchangeSDK as it should not be really used and only rely on the url passed when creating the exchangeSDK. 

This a regression for swap